src/quantuminspire/projectq/backend_qx.py

```diff
@@ -303,7 +395,8 @@ def _store(self, cmd: Command) -> None:
             self._measured_ids += [logical_id]
             # do not add the measurement statement when fsp is possible
             if not self._full_state_projection:
-                self.qasm += "\nmeasure q[{}]".format(sim_qubit_id)
+                if self._is_simulation_backend:
+                    self.qasm += "\nmeasure q[{}]".format(sim_qubit_id)
```

Comment:
Can you make this an f-string?

Reply:
I changed all strings to f strings in this file.